### PR TITLE
Closes #26502: add isLocal parameter for history openItem telemetry event

### LINKED ISSUE #26502: Telemetry: Open History Item
Questions to answer:

- How often is a local history item opened
- How often is a synced history item opened

Proposed metrics:

- Add a label to the existing history.open_item that identifies what type of history item it was

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21315)

### PR DESCRIPTION
**Pull request by mavduevskiy**
Closes #26502 

Adding a new parameter to an existing event.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26502

### CODE REVIEW

**Review by Amejia481 — APPROVED** [review]
Looks good, just new to address the failing test.

```

2022-08-17T06:24:13.9801956Z   TEST: WHEN record is called THEN telemetry is recorded with the appropriate label
2022-08-17T06:24:13.9803029Z     W/PackageParser: Unknown element under <manifest>: queries at /home/runner/work/fenix/fenix/app/build/intermediates/apk_for_local_test/debugUnitTest/apk-for-local-test.ap_ Binary XML file line #40
2022-08-17T06:24:13.9803528Z     I/glean/MetricsPingSched: New MetricsPingSched
2022-08-17T06:24:13.9803834Z     I/glean/MetricsPingSched: New MetricsPingSched
2022-08-17T06:24:13.9804148Z     I/glean/MetricsPingSched: MetricsPingSched.schedule
2022-08-17T06:24:13.9804589Z     D/glean/MetricsPingSched: The 'metrics' ping was last sent on 1436572800000
2022-08-17T06:24:13.9805284Z     I/glean/MetricsPingSched: The 'metrics' ping was already sent today, 1660717453961.
2022-08-17T06:24:13.9805739Z     D/glean/MetricsPingSched: Scheduling the 'metrics' ping in 77746039ms
2022-08-17T06:24:14.0790570Z     I [StartupTypeTelemetry] Recorded start up: cold_main
2022-08-17T06:24:14.1791730Z   FAILURE
2022-08-17T06:24:14.3791645Z 
2022-08-17T06:24:14.3845380Z java.lang.AssertionError: expected:<1> but was:<null>
2022-08-17T06:24:14.3845956Z 	at org.junit.Assert.fail(Assert.java:88)
2022-08-17T06:24:14.3846689Z 	at org.junit.Assert.failNotEquals(Assert.java:834)
2022-08-17T06:24:14.3847174Z 	at org.junit.Assert.assertEquals(Assert.java:118)
2022-08-17T06:24:14.3847885Z 	at org.junit.Assert.assertEquals(Assert.java:144)
2022-08-17T06:24:14.3848630Z 	at org.mozilla.fenix.perf.StartupTypeTelemetryTest.WHEN record is called THEN telemetry is recorded with the appropriate label(StartupTypeTelemetryTest.kt:96)
2022-08-17T06:24:14.3849655Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-08-17T06:24:14.3850410Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-08-17T06:24:14.3851425Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-08-17T06:24:14.3852097Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-08-17T06:24:14.3852877Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
2022-08-17T06:24:14.3853702Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-08-17T06:24:14.3854916Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
2022-08-17T06:24:14.3855578Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-08-17T06:24:14.3856431Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-08-17T06:24:14.3857003Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
2022-08-17T06:24:14.3857715Z 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
2022-08-17T06:24:14.3858287Z 	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:591)
2022-08-17T06:24:14.3859444Z 	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:274)
2022-08-17T06:24:14.3860118Z 	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:88)
2022-08-17T06:24:14.3860922Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-08-17T06:24:14.3861527Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2022-08-17T06:24:14.3862383Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2022-08-17T06:24:14.3862910Z 	at java.base/java.lang.Thread.run(Thread.java:829)
2022-08-17T06:24:14.3864257Z 	Suppressed: org.robolectric.android.internal.AndroidTestEnvironment$UnExecutedRunnablesException: Main looper has queued unexecuted runnables. This might be the cause of the test failure. You might need a shadowOf(Looper.getMainLooper()).idle() call.
2022-08-17T06:24:14.3865090Z 
```

**Comment by mavduevskiy** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
  * This data will help us understand if users are more interacting with local or synced history items.
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * We are considering allowing users accessing local/synced history separately. The hypothesis being the browser usage on a mobile device might be different from desktop version, so synced items might flood the history list while what user is actually interested is going through the local history.
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No. 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
  * No new entries, parameters only.
6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix 
7. How long will this data be collected? Choose one of the following:
  * Up until version 115, with the option to renew at that point.
8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean and Glam. 
11. Where do you intend to share the results of your analysis?
  *  Only on Glean, Glam, and with mobile teams. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

File: app/metrics.yaml [hunk 1]
```diff
@@ -3041,6 +3041,11 @@ history:
     type: event
     description: |
       A user opened a history item
+    extra_keys:
+      is_remote:
+        description: |
+          Shows if the item was opened locally or on other devices.
+        type: boolean
```
Comment (rocketsroger) [inline app/metrics.yaml:3048]: nit: let's move `type` above `description`

File: app/metrics.yaml [hunk 2]
```diff
@@ -3041,6 +3041,11 @@ history:
     type: event
     description: |
       A user opened a history item
+    extra_keys:
+      is_remote:
+        description: |
+          Shows if the item was opened locally or on other devices.
```
Comment (rocketsroger) [inline app/metrics.yaml:3047]: Suggestion: True if the history item is remote, otherwise false.

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * This data will help us understand if users are more interacting with local or synced history items.
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
>     * We are considering allowing users accessing local/synced history separately. The hypothesis being the browser usage on a mobile device might be different from desktop version, so synced items might flood the history list while what user is actually interested is going through the local history.
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * There are no other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No.
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Extras for opened history: user has opened item synced from other devices or browed locally. 	Category 2 - interaction data 	#26502
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix%E2%80%A8)
> 
> 
>     7. How long will this data be collected? Choose one of the following:
> 
> 
>     * Indefinitely.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean and Glam.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean, Glam, and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

Permanent, monitoring email updated.

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Review by rocketsroger — APPROVED** [review]

**Pull request merged**